control/HTTPRequest.php

```diff
@@ -655,15 +655,19 @@ public function allParsed() {
 	 * @return string
 	 */
 	public function getIP() {
+		$ip = false;
 		if (!empty($_SERVER['HTTP_CLIENT_IP'])) {
 			//check ip from share internet
-			return $_SERVER['HTTP_CLIENT_IP'];
+			$ip = $_SERVER['HTTP_CLIENT_IP'];
 		} elseif (!empty($_SERVER['HTTP_X_FORWARDED_FOR'])) {
 			//to check ip is pass from proxy
-			return  $_SERVER['HTTP_X_FORWARDED_FOR'];
-		} elseif(isset($_SERVER['REMOTE_ADDR'])) {
-			return $_SERVER['REMOTE_ADDR'];
+			$ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
 		}
+		if(!$ip || !filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)) {
+			//if no other forwarding ip is found, invalid, or internal ip address
+			$ip = $_SERVER['REMOTE_ADDR'];
```

Comment:
@dhensby Yes, for instance I have a transparent squid proxy inside my own network I use for testing sometimes. This automatically adds `HTTP_X_FORWARDED_FOR` with my internal IP, and the current getIP() returns that rather than my external one. The other point to mention is that `HTTP_X_FORWARDED_FOR` is easily spoofed, and thus could be set to anything (other than an IP address as far as I know). This simply resorts back to the `REMOTE_ADDR` if it doesn't validate as an external IP address.

@kinglozzer That's a very good point, and yes, I overlooked that!

I'll modify my patch ...